src/components/RenderCanvas.vue

```diff
@@ -873,11 +929,13 @@ function onRun(runCompiledCode: CompiledPlayground) {
             if (compiler == null) {
                 throw new Error("Could not get compiler");
             }
-            if (compiler.shaderType !== null && !abortRender) {
+            if (compiler.shaderType !== null && !abortRender && !pauseRender.value) {
                 const keepRendering = await execFrame(timeMS, compiler?.shaderType || null, compiledCode, firstFrame);
                 firstFrame = false;
                 return keepRendering;
             }
+            if (!abortRender) return true;
+            
             return false;
```

Comment:
I think there's a more readable way to implement this logic
```suggestion
            if (abortRender) return false;
            if (pauseRender.value) return true;

            if (compiler.shaderType === null) {
                // handle this case
            }
            const keepRendering = await execFrame(timeMS, compiler?.shaderType || null, compiledCode, firstFrame);
            firstFrame = false;
            return keepRendering;
```